src/core/player.ts

```diff
@@ -35,19 +41,32 @@ export class Player extends Entity {
     if (input.down) { inputVel.y += step; }
     if (input.left) { inputVel.x -= step; }
     if (input.right) { inputVel.x += step; }
-
-    player.orientation = input.angle;
-    player.pos.x += inputVel.x;
-    player.pos.y += inputVel.y;
+    
 
     let events: any = [];
+    
+    events.push(EventFactory.createEvent('MOVEMENT', player.id, null, {
+      angle: input.angle,
+      xVel: inputVel.x,
+      yVel: inputVel.y, 
+    }));
+
+
      // TODO: Investigate whether this Date.now business messes up with server
     if (input.fired && player.lastFire + player.gunCooldown < Date.now()) {
       player.lastFire = Date.now();
-      events.push(EventFactory.createEvent('SPAWN_BULLET', player.id, null));
+      events.push(EventFactory.createEvent('SPAWN_BULLET', player.id, null, {}));
```

Comment:
You already dealt with this case of `{}` as the default last param in `createEvent` right?